index.js

```diff
@@ -0,0 +1,114 @@
+const commandLineArgs = require('command-line-args');
+const fs = require('fs');
+// const mongoose = require('mongoose');
+const readline = require('readline');
+
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+const optionDefinitions = [
+    { name: 'src', type: String, multiple: false, defaultOption: true },
+    { name: 'threshold', type: Number }
+]
+
+const options = commandLineArgs(optionDefinitions);
+const W = options.threshold + 1
+
+
+function logObject(o) {
+    console.log(`logObject length: ${o.length}`)
+    for (let i = 0; i < o.length; i++) {
+        process.stdout.write(`${o[i].id} `);
+    }
+    console.log();
+}
+function checkDups(m) {
+    const dups = [];
+    for (let i = 0; i < m.length; i++) {
+        const id = m[i].id;
+        if (dups.indexOf(id) >= 0)
+            console.log(`dup id ${id}`);
+        else
+            dups.push(id);
+    }
+    if (dups.length == m.length) console.log('no dups');
+}
+function logRow(row, rowNumber) {
```

Comment:
This and other logging functions are a nice touch.

Reply:
thanks